kolibri/core/analytics/measurements.py

```diff
@@ -0,0 +1,173 @@
+
+    return (active_sessions, active_users, active_users_minute)
+
+
+def get_channels_usage_info():
+    """
+    Scan the channels Kolibri has installed, getting information on how many times
+    their resources have been accessed and how long they have been used
+    :returns: List containing namedtuples, with each channel: id, name, accesses and time spent
+    """
+    channels_info = []
+    ChannelsInfo = namedtuple('ChannelsInfo', 'id name accesses time_spent')
+
+    try:
+        connection.ensure_connection()
+        channels = ChannelMetadata.objects.values('id', 'name')
+        channel_stats = ContentSessionLog.objects.values('channel_id').annotate(time_spent=Sum('time_spent'),
+                                                                                total=Count('channel_id'))
+        for channel in channels:
+            stats = channel_stats.filter(channel_id=channel['id'])
+            if stats:
+                channels_info.append(ChannelsInfo(id=channel['id'], name=channel['name'],
+                                                  accesses=str(stats[0]['total']),
+                                                  time_spent='{:.2f} s'.format(stats[0]['time_spent'])))
+            else:
+                channels_info.append(ChannelsInfo(id=channel['id'], name=channel['name'],
+                                                  accesses='0', time_spent='0.00 s'))
+    except OperationalError:
+        print('Database unavailable, impossible to retrieve channels usage info')
+    return channels_info
+
+
+def get_requests_info():
+    """
+    Returns timing information on some Kolibri pages that can be hit without credentials
+    :returns: tuple of strings containing time in seconds when requesting
+              - Kolibri homepage
+              - Kolibri recommended channels
+              - Kolibri channels list
+    """
+    def format_url(url, base_url):
+        formatted = '{base_url}{url}&contentCacheKey={cache}'.format(base_url=base_url, url=url, cache=time.time())
+        return formatted
+
+    _, port = get_kolibri_process_info()
+    if port:
+        base_url = 'http://localhost:{}'.format(port)
+        homepage_time = '{:.2f} s'.format(requests.get(base_url).elapsed.total_seconds())
+        recommended_url = format_url('/api/content/contentnode_slim/popular/?by_role=true', base_url)
+        recommended_time = '{:.2f} s'.format(requests.get(recommended_url).elapsed.total_seconds())
+        channels_url = format_url('/api/content/channel/?available=true', base_url)
+        channels_time = '{:.2f} s'.format(requests.get(channels_url).elapsed.total_seconds())
+
+    return (homepage_time, recommended_time, channels_time)
+
+
+def get_process_pid():
+    """
+    Returns the pid of the command in execution
+    :returns: Integer number with the pid of this command
+    """
+    this_command = psutil.Process()
+    return this_command.pid
```

Comment:
You should be able to avoid the call to `psutil` here by using the `.pid` file as a cross-platform alternative? Is there an issue with that? In that case, it's pretty bad since we rely on the `.pid` file for other purposes...

Reply:
This is the pid of the command being executed, not from Kolibri. For kolibry I rely on the implementation we already have.

Reply:
This change won't save any code in pskolibri. If you look at the Process class in __init__.py, it uses os.getpid() to get the pid when instantiating a new Process if the pid is not provided. It's needed to use some of the methods this class has.

In any case, I like  your suggestion, the code I wrote there is a clear case of overengineering. I am changing it.